qute.jdt/com.redhat.qute.jdt/src/main/java/com/redhat/qute/jdt/internal/extensions/quarkus/InjectNamespaceResolverSupport.java

```diff
@@ -68,6 +69,11 @@ protected void processAnnotation(IJavaElement javaElement, IAnnotation annotatio
 
 			IType type = (IType) javaElement;
 			String named = getNamed(type);
+
+			// Filter any occurrences of @Stereotype usage with @Named

```

Comment:
I think `IType.isInterface() && IType.isAnnotation()` could probably be used to filter out `@interface`.

Reply:
@mkouba it seems that `inject` in Qute template doesn't work when:

 * class in an abstract class
 * record have some parameters
 * type is @interface

Is it a CDI specification or is it a Qute rule? Do you know where we can find some rules when some Java type are annotated with @Named but it cannot be used with `inject` namespace resolver in Qute template?

Reply:
> * class in an abstract class

An abstract class is not a bean, i.e. you cannot `@Inject` it or use in a template.

> record have some parameters

Hm, I'm not quite sure how records are treated by the CDI...

> type is @interface

An interface is not a bean.

The CDI rules: https://jakarta.ee/specifications/cdi/2.0/cdi-spec-2.0.html#what_classes_are_beans